src/tracing/vizceral_json_file_creator_nodes.coffee

```diff
@@ -0,0 +1,176 @@
+  getTopicToChannelbyId = (success) ->
+    traceIdToTopicChannel = {}
+    #for value, index in success
+    for key, value in success
+      dataArray = key.split('|')[0..1].join('>')
+      key_trace = key.split('|')[2]
+      if key_trace of traceIdToTopicChannel
+        traceIdToTopicChannel[key_trace] = traceIdToTopicChannel[key_trace] + "," + dataArray
+      else
+        traceIdToTopicChannel[key_trace] = dataArray
+    traceIdToTopicChannel
+
+  # This function reduces the map by bringing in count for repetition
+  reduceMap = (traceIdToTopicChannel) ->
+    reducedMap = {}
+    for key, value of traceIdToTopicChannel
+      arrayOfTC = value.split(',')
+      for item, index in arrayOfTC
+        if arrayOfTC[index] of reducedMap
+          reducedMap[item]++
+        else
+          reducedMap[item] = 1
+    reducedMap
+
+  # This function prepares output list src>targer>#count
+  getSourceToTarget = (reducedMap) ->
+    size = _.keys(reducedMap).length
+    outputList = []
+    for value, index of reducedMap
+      [src, trg] = value.split('>')
+      source = src.split(':')[0]
+      target = trg
+      outputList.push(source+'>'+target+'#'+reducedMap[value])
+    outputList
+
+  # Ths function prepares the nodeList for adding to JSON finally
+  getNodeList = (srcToTarget) ->
+    resultNodes = []
+    resultNodes.push('narrows')
+    for row in srcToTarget
+      [first, rest] = row.split('>')
+      if first not in resultNodes
+        resultNodes.push(first)
+      if rest.split('#')[0] not in resultNodes
+        resultNodes.push(rest.split('#')[0])
+    for node in resultNodes
+      name: node
+      class: 'normal'
+
+  # Ths function prepares the source to target mapping for adding to JSON finally
+  getSourceToTargetList = (srcToTarget, entryhead, mapOfTopicToCount) ->
+    mapOfEntryAndVal = {}
+    srcToTargetList = for index, node of srcToTarget
+      val = node.split('>')
+      targetNode = val[1].split('#')
+      if val[0] in entryhead
+        if val[0] of mapOfEntryAndVal
+          count = mapOfEntryAndVal[val[0]] + targetNode[1] * 100
+          mapOfEntryAndVal[val[0]] = count
+        else
+          mapOfEntryAndVal[val[0]] = targetNode[1] * 100
+      source: val[0]
+      target: targetNode[0]
+      metrics:
+        danger: mapOfTopicToCount[targetNode[0]] * 100
+        normal: targetNode[1] * 100
+      class: 'normal'
+    srcToTargetEntry = for node, index of mapOfEntryAndVal
+      source: 'narrows'
+      target: node
+      metrics:
+        danger: 0
+        normal: index
+    srcToTargetList.concat(srcToTargetEntry)
+
+  # This method find all the entrypoint narrows
+  findEntryHeads = (srcToTarget) ->
+    endNodesList = []
+    startNodeList = []
+    for index, node of srcToTarget
+      if node in endNodesList
+        continue
+      else
+        endNodesList.push(node.split('>')[1].split('#')[0])
+    for index, node of srcToTarget
```

Comment:
actually no I can not. I first prepare just the endNode list and then subtract the list from the entire list to find the entry point nodes to which narrows will get connected to.